recipe/meta.yaml

```diff
@@ -59,23 +68,24 @@ requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - {{ compiler('cuda') }}                        # [cuda_compiler_version != "None"]
+    - {{ compiler('cuda') }}                        # [(cuda_compiler_version or "").startswith("12")]
     - cmake
     - git
     - ninja
     - pkgconfig
   host:
     # NOTE: Without cuda-version, we are installing cuda-toolkit 11.8 instead of 11.2!
-    - cuda-version     {{ cuda_compiler_version }}  # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
+    - cudatoolkit      {{ cuda_compiler_version }}  # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
     - cuda-cudart-dev  {{ cuda_compiler_version }}  # [(cuda_compiler_version or "").startswith("12")]
     - libcublas-dev    {{ cuda_compiler_version }}  # [(cuda_compiler_version or "").startswith("12")]
 
     - blas-devel * *{{ blas_impl }}                 # [not osx and cuda_compiler_version == "None" and blas_impl == "mkl"]
     - mkl-devel {{ mkl }}                           # [not osx and cuda_compiler_version == "None" and blas_impl == "mkl"]
   run:
-    - cuda-version {{ cuda_compiler_version }}      # [cuda_compiler_version != "None" and not (cuda_compiler_version or "").startswith("12")]
```

Comment:
This would give us `cudatoolkit 12.3|12.3.*` and `__cuda >=12` for free, see https://conda-metadata-app.streamlit.app/?q=pkgs%2Fmain%2Fnoarch%2Fcuda-version-12.3-h55a0123_2.conda.

Reply:
I think the cuda-version feedstock also needs to be looked at more closely. I opened a ticket, feel free to comment/discuss if so inclined :) https://anaconda.atlassian.net/browse/PKG-4305